src/sap.f/src/sap/f/DynamicPage.js

```diff
@@ -699,13 +699,23 @@ sap.ui.define([
 		bUseAnimations = sAnimationMode !== Configuration.AnimationMode.none && sAnimationMode !== Configuration.AnimationMode.minimal;
 
 		if (exists(this.$contentFitContainer)) {
-			this.$contentFitContainer.toggleClass("sapFDynamicPageContentFitContainerFooterVisible", bShow);
+			if (this.getDomRef()) {
+				var oContentFitContainer = this.getDomRef().querySelector("[id$='contentFitContainer']");
```

Comment:
This is pretty much the same thing as is today with jQuery. Have a look at how this.$contentFitContainer is set on line [1792](https://github.com/SAP/openui5/blob/6d0303f509537763dd321a3d4f48a10be23bf998/src/sap.f/src/sap/f/DynamicPage.js#L1792):

`this.$contentFitContainer = this.$("contentFitContainer");`

Today, jQuery looks for any child id which contains "contentFitContainer" (at any position, it may not be a suffix) with `this` as the parent. The change doesn't use contains, it ensures child id ends with "contentFitContainer", it has to be a suffix. The query is always performed in the controller context making it very specific and reducing the risk of getting unexpected results.

Reply:
Sorry, it isn't the same and definitely not safer. `this.$("contentFitContainer")` translates to 

```js
  jQuery(document.getElementById(this.getId() + "-" + "contentFitContainer"));
```

which looks for a very specific ID. See [Element.js#L423-460](https://github.com/SAP/openui5/blob/master/src/sap.ui.core/src/sap/ui/core/Element.js#L423-L460).

`querySelector` returns the first DOM element in document order that matches the selector. And as sketched above, this is NOT safe with a pure suffix search (in the general case).
 
And I would even expect the `querySelector` to be less efficient. `getElementById` is (hopefully) a simple lookup, whereas `querySelector` has to search and match DOM elements. Can be optimised by the browser, too. But I wouldn't bet on it that every possible kind of selector has been optimised.

Reply:
Oh, my bad! I haven't seen `Element.prototype.$` implementation before. So, for these cases, just executing `this.getDomRef('someSuffix')` should suffice as it'd get by full ID and not return a jQuery object.

Performance wise, `getElementById` is ~2x  faster than `querySelector`. However, it doesn't mean `querySelector` is slow, it executes +2M ops/sec.
https://www.measurethat.net/Benchmarks/Show/2488/0/getelementbyid-vs-queryselector